core/src/price_estimation.rs

```diff
@@ -123,22 +129,20 @@ impl PriceEstimating for PriceOracle {
 
             let mut tokens = Tokens::new();
             for token_id in token_ids_to_price {
-                let price = prices.get(&token_id).copied();
-                let token_info =
-                    if let Ok(base_info) = self.token_info_fetcher.get_token_info(token_id).await {
-                        Some(TokenInfo {
-                            external_price: price.unwrap_or(base_info.external_price),
-                            ..base_info.into()
-                        })
-                    } else if let Some(price) = price {
-                        Some(TokenInfo {
-                            alias: None,
-                            decimals: None,
-                            external_price: price,
-                        })
-                    } else {
-                        None
```

Comment:
This PR actually changes this logic in an undesired way. We only use the external price of price is not available (I though we always use the external price if there is custom token info).